misk/src/main/kotlin/misk/hibernate/ReflectionQueryFactory.kt

```diff
@@ -293,6 +294,37 @@ internal class ReflectionQuery<T : DbEntity<T>>(
               }
             }
           }
+          Operator.IN -> {
+            val parameter = function.parameters[1]
+
+            val argToCollection: (Any) -> Collection<Any?> = when {
+              parameter.isVararg -> { arg -> (arg as Array<*>).toList() }
+              parameter.isAssignableTo(Collection::class) -> { arg -> (arg as Collection<*>) }
+              else -> {
+                errors.add("${function.name}() parameter must be a vararg or a collection")
+                return
+              }
+            }
+
+            object : QueryMethodHandler {
+              override fun invoke(query: ReflectionQuery<*>, args: Array<out Any>) {
+                query.addConstraint { root, builder ->
+                  val collection = argToCollection(args[0])
+                  if (collection.isEmpty()) {
+                    // The JDBC API forbids empty IN clauses so we use `WHERE 1 = 0` instead to
```

Comment:
anyway to not execute the query?

Reply:
Not really. This code will eventually compose and you might say `WHERE name IN () OR release_date < '2018-01-01'` and the IN clause reduces to false but not to an empty query.

Reply:
true true